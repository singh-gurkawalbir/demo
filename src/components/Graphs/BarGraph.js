import {
  BarChart,
  Bar,
  CartesianGrid,
  Legend,
  XAxis,
  YAxis,
  Tooltip,
  ResponsiveContainer,
} from 'recharts';
import './Bar.css';
import React from 'react';

function BarGraph({ data, onChange }) {
  // const CustomTooltip = ({ active, payload, label }) => {
  //     if (active && payload && payload.length) {
  //       return (
  //         <div className="custom-tooltip">
  //           <p className="label">{`${label}`}</p>
  //           <hr />
  //           <p>{`${payload[0].value}/150`}</p>
  //           <p>{`${payload[1].value}/150`}</p>
  //           <p>{"A and B represent the marks obtained"}</p>
  //         </div>
  //       );
  //     }

  //     return null;
  //   };

  // eslint-disable-next-line react/no-array-index-key
  const color = ['#D93535', '#05B39C'];
  // eslint-disable-next-line react/no-array-index-key
  const renderedBar = data.ids.Plots.map((plot, index) => <Bar key={index} dataKey={plot} stackId="a" fill={color[index]} />);

  return (
    <ResponsiveContainer minWidth={300} minHeight={220}>
      <BarChart
        width={500}
        height={400}
        data={data.values}
        onClick={onChange}
        margin={{ top: 20, right: 30, left: 20, bottom: 5 }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey={data.ids.XAxis} />
        <YAxis />
        <Tooltip data-testid="tooltip" />
        <Legend />
        {renderedBar}
      </BarChart>
    </ResponsiveContainer>
  );
}

export default BarGraph;
