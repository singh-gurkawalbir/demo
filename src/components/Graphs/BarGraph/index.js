import {
  BarChart,
  Bar,
  CartesianGrid,
  Legend,
  XAxis,
  YAxis,
  Tooltip,
  ResponsiveContainer,
} from 'recharts';
import React from 'react';

function BarGraph({ data, color }) {
  // eslint-disable-next-line react/no-array-index-key
  const renderedBar = data.ids.Plots.map((plot, index) => <Bar key={index} dataKey={plot} stackId="a" fill={color[index]} />);

  return (
    <ResponsiveContainer minWidth={300} minHeight={220}>
      <BarChart
        width={500}
        height={400}
        data={data.values}
        // onClick={onChange}
        margin={{ top: 20, right: 30, left: 20, bottom: 5 }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey={data.ids.XAxis} />
        <YAxis />
        <Tooltip data-testid="tooltip" />
        <Legend />
        {renderedBar}
      </BarChart>
    </ResponsiveContainer>
  );
}

export default BarGraph;
