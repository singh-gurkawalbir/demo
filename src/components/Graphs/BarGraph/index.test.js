import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import BarGraph from '.';

global.ResizeObserver = require('resize-observer-polyfill');

jest.mock < typeof import('recharts') > ('recharts', () => {
  const OriginalModule = jest.requireActual('recharts');

  return {
    ...OriginalModule,
    ResponsiveContainer: ({ children }) => (
      <OriginalModule.ResponsiveContainer width={300} height={300}>
        {children}
      </OriginalModule.ResponsiveContainer>
    ),
  };
});

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

describe('BarGraph', () => {
  const data = {
    ids: {
      XAxis: 'xaxis',
      Plots: ['plot1', 'plot2'],
    },
    values: [
      {
        xaxis: 'Jan',
        plot1: 100,
        plot2: 50,
      },
      {
        xaxis: 'Feb',
        plot1: 70,
        plot2: 90,
      },
    ],
  };

  const color = ['#8884d8', '#82ca9d'];

  test('Tooltips are not visible by default', () => {
    const { container } = render(<BarGraph data={data} color={color} />);

    // eslint-disable-next-line jest/valid-expect
    expect(container.querySelector('recharts-tooltip-wrapper')).not.toBeVisible;
  });

  test('Can create a test snapshot', async () => {
    const { container } = render(<BarGraph data={data} color={color} />);

    expect(container).toMatchSnapshot();
  });

  test('A cursor does not exist', async () => {
    const { container } = render(<BarGraph data={data} color={color} />);

    expect(container.querySelector('recharts-tooltip-cursor')).toBeNull();
  });

  test('Correct values are displayed on tooltips', () => {
    const { container } = render(<BarGraph data={data} color={color} />);
    const element = container.querySelector('recharts-tooltip-wrapper');

    // eslint-disable-next-line jest/valid-expect-in-promise
    userEvent.hover(element)
      .then(() => {
        const tooltipElements = container.querySelectorAll('recharts-tooltip-item-value');

        expect(tooltipElements[0].props().value).toBe(100);
        expect(tooltipElements[1].props().value).toBe(70);
      });
  });
});
