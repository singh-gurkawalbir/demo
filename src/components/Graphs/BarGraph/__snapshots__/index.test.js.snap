// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarGraph Can create a test snapshot 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 100%; min-width: 300px; min-height: 220px;"
  />
</div>
`;
